discovery_engine_core/bindings/src/types/document/user_reaction.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Xayn AG
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, version 3.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+//! FFI functions for handling [`UserReaction`] fields.
+
+use core::document::UserReaction;
+
+type IntRepr = u8;
+
+/// Initializes an [`UserReaction`] field at given place.
+///
+/// # Safety
+///
+/// It must be valid to write an [`UserReaction`] to given pointer.
+#[no_mangle]
+pub unsafe extern "C" fn init_user_reaction_at(place: *mut UserReaction, reaction: IntRepr) -> u8 {
+    let mut ok = true;
+    let reaction = match reaction {
+        0 => UserReaction::Neutral,
+        1 => UserReaction::Positive,
+        2 => UserReaction::Negative,
+        _ => {
+            ok = false;
+            UserReaction::default()
+        }
+    };
+    unsafe {
+        place.write(reaction);
+    }
+    ok as u8
+}
+
+/// Gets the int representation of an [`UserReaction`].
+///
+/// # Safety
+///
+/// The pointer must point to a valid [`UserReaction`] instance.
+#[no_mangle]
+pub unsafe extern "C" fn get_user_reaction(reaction: *mut UserReaction) -> IntRepr {
+    let reaction = unsafe { &*reaction };
+    match reaction {
```

Comment:
this could be inlined without the readability being reduced
```suggestion
    match unsafe { &*reaction } {
```